src/main/java/com/spring/studentmanagement/services/UserServiceImpl.java

```diff
@@ -34,7 +39,13 @@ public AppUser getUserById(Long userId) throws UserNotFoundExceptions {
         log.info("Trying to getUserById with param: userId = {}", userId);
 
         return this.userRepository.findById(userId).orElseThrow(() -> new UserNotFoundExceptions("User not found!"));
+    }
 
+    @Override
+    public AppUser saveUser(AppUser user) {
+        String hashedPassword = new BCryptPasswordEncoder().encode(user.getPassword());
```

Comment:
We need to create a Bean of Password encoder instead of creating a new instance each time we save a new user. This can lead to memory consumption.
Create a new class SecurityConfig under the security package.
Annotate the class with @Configuration and there you can create a new bean of PasswordEncodder.